azure/services/scalesets/scalesets.go

```diff
@@ -475,6 +452,123 @@ func (s *Service) buildVMSSFromSpec(ctx context.Context, vmssSpec azure.ScaleSet
 		},
 	}
 
+	// Use custom NIC definitons in VMSS if set
+	if len(vmssSpec.NetworkInterfaces) > 0 {
+		nicConfigs := []compute.VirtualMachineScaleSetNetworkConfiguration{}
+		for i, n := range vmssSpec.NetworkInterfaces {
+			nicConfig := compute.VirtualMachineScaleSetNetworkConfiguration{}
+			nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties = &compute.VirtualMachineScaleSetNetworkConfigurationProperties{}
+			nicConfig.Name = to.StringPtr(vmssSpec.Name + "-" + strconv.Itoa(i))
+			if n.ID != "" {
+				nicConfig.ID = &n.ID
+			} else {
+				if to.Bool(n.AcceleratedNetworking) {
+					nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = to.BoolPtr(true)
+				} else {
+					nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.EnableAcceleratedNetworking = to.BoolPtr(false)
+				}
+				if n.PrivateIPConfigs == 0 && n.PublicIPConfigs == 0 {
+					nicConfig.VirtualMachineScaleSetNetworkConfigurationProperties.IPConfigurations = &[]compute.VirtualMachineScaleSetIPConfiguration{
+						{
+							Name: to.StringPtr(vmssSpec.Name + "-" + strconv.Itoa(i)),
+							VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
+								Subnet: &compute.APIEntityReference{
+									ID: to.StringPtr(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, n.SubnetName)),
+								},
+								Primary:                         to.BoolPtr(true),
+								PrivateIPAddressVersion:         compute.IPVersionIPv4,
+								LoadBalancerBackendAddressPools: &backendAddressPools,
+							},
+						},
+					}
+				} else {
+					ipconfigs := []compute.VirtualMachineScaleSetIPConfiguration{}
+					for j := 0; j < n.PublicIPConfigs; j++ {
+						ipconfig := compute.VirtualMachineScaleSetIPConfiguration{
+							Name: to.StringPtr(fmt.Sprintf("public-ipConfig%v", j)),
+							VirtualMachineScaleSetIPConfigurationProperties: &compute.VirtualMachineScaleSetIPConfigurationProperties{
+								PrivateIPAddressVersion: compute.IPVersionIPv4,
+								Subnet: &compute.APIEntityReference{
+									ID: to.StringPtr(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, n.SubnetName)),
+								},
+								PublicIPAddressConfiguration: &compute.VirtualMachineScaleSetPublicIPAddressConfiguration{},
+							},
+						}
+						if j == 0 {
+							ipconfig.Primary = to.BoolPtr(true)
+							if i == 0 {
```

Comment:
Similar to what @jackfrancis said, is there any other identifier we can use besides the index to check for this case?